### PR TITLE
Upgrade azure libraries

### PR DESCRIPTION
**Pull request by sephii**
_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

The currently used releases of the azure libraries used in tests are 2-3 years old. This PR ensures the code works with the latest version of these libraries.

Note that this change is required to get Wagtail as a Nix package, so it would be really nice if this can be integrated soon so I can move forward with the Nix package. ;)

### CODE REVIEW

File: wagtail/contrib/frontend_cache/backends.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def _get_subscription_id(self):
 
     def _get_client_kwargs(self):
         return {
-            "credentials": self._get_credentials(),
+            "credential": self._get_credentials(),
```
Comment (gasman) [inline wagtail/contrib/frontend_cache/backends.py:296]: We'll need to consider how this change affects end-users. Presumably this code has been incompatible with azure-mgmt-cdn >=10 up to this point, and this change will break things for a user who has the dependency pinned to <10. At the very least we need to document the version requirements in the docs and release notes - ideally we'd identify the version in use and pass the appropriate kwarg.

**Review by gasman — CHANGES_REQUESTED** [review]

File: wagtail/contrib/frontend_cache/backends.py (hunk 1 above)
Comment (sephii) [reply wagtail/contrib/frontend_cache/backends.py:296]: Good point, I’ll add a version check to make sure the change is backwards-compatible. Should I also add a deprecation warning, so we can remove the compatibility code in a future release?

Comment (gasman) [reply wagtail/contrib/frontend_cache/backends.py:296]: @sephii Yes please!

Comment (sephii) [reply wagtail/contrib/frontend_cache/backends.py:297]: Here you go! Let me know if it matches what you had in mind (and feel free to add any mention about the change in the release notes).

**Review by gasman — APPROVED** [review]
All looks good to me, thanks!

**Comment by gasman** [comment]
Merged in 1da5b97d513d671724905d67ce54c84b86433065.

**Pull request closed without merging**